test/test_integration_ssl.rb

```diff
@@ -81,12 +94,23 @@ def stop_server
 
   def test_ssl_run
     body = nil
-    start_server
+    start_server("#{BASE} bin/puma -C #{generate_config(@default_config)}")
     @http.start do
       req = Net::HTTP::Get.new '/', {}
       @http.request(req) { |resp| body = resp.body }
     end
     assert_equal 'https', body
     stop_server
   end
+
+  def test_ssl_run_with_localhost_authority
+    body = nil
+    start_server("#{BASE} bin/puma -C #{generate_config(@localhost_config)}")
+    @http.start do
+      req = Net::HTTP::Get.new '/', {}
+      @http.request(req) { |resp| body = resp.body }
+    end
+    assert_equal 'https', body
+    stop_server
+  end unless ::Puma::IS_JRUBY
```

Comment:
I would prefer `skip ...` or `skip_if` over so we can see from the test output that we are lacking support in JRuby. I also think we need to skip Windows? ([reference](https://github.com/socketry/localhost/blob/664d28071e458375f6f378c48897820a84658d40/.github/workflows/development.yml#L12-L20))

Reply:
My bad, looks like Windows works: https://github.com/puma/puma/runs/3796685009?check_suite_focus=true#step:9:373 

I was thinking `local_certificates_path = File.expand_path("~/.localhost")` wouldn't work in Windows, but that code is probably never exercised because the localhost gem can respond to the `_path` methods